docs/resources/instancev2.md

```diff
@@ -0,0 +1,524 @@
+    port_range_max = 65535
+  }
+
+  security_group_rules {
+    direction      = "ingress"
+    ethertype      = "IPv4"
+    protocol       = "tcp"
+    port_range_min = 22
+    port_range_max = 22
+  }
+
+  security_group_rules {
+    direction      = "ingress"
+    ethertype      = "IPv4"
+    protocol       = "tcp"
+    port_range_min = 80
+    port_range_max = 80
+  }
+
+  security_group_rules {
+    direction      = "ingress"
+    ethertype      = "IPv4"
+    protocol       = "tcp"
+    port_range_min = 443
+    port_range_max = 443
+  }
+
+}
+
+resource "gcore_instancev2" "instance" {
+  flavor_id     = "g1-standard-2-4"
+  name          = "my-instance"
+  keypair_name  = "my-keypair"
+
+  volume {
+    source     = "existing-volume"
+    volume_id  = gcore_volume.boot_volume.id
+    boot_index = 0
+  }
+
+  interface {
+    type = "external"
+    name = "my-external-interface"
+
+    security_groups = [gcore_securitygroup.web_server_security_group.id]
+  }
+
+  project_id = data.gcore_project.project.id
+  region_id  = data.gcore_region.region.id
+}
+```
+
+
+#### Creating Windows instance with two users
+
+This example shows how to create a Windows instance with two users. The second user is added by using
+the userdata feature to automate the creation process.
+
+userdata:
+```userdata
+<powershell>
+# Be sure to set the username and password on these two lines. Of course this is not a good
+# security practice to include a password at command line.
+$User = "SecondUser"
+$Password = ConvertTo-SecureString "s3cR3tP@ssw0rd" -AsPlainText -Force
+New-LocalUser $User -Password $Password
+Add-LocalGroupMember -Group "Remote Desktop Users" -Member $User
+Add-LocalGroupMember -Group "Administrators" -Member $User
+</powershell>
+```
+
+```terraform
+data "gcore_image" "windows" {
+  name       = "windows-server-2022"
+  region_id  = data.gcore_region.region.id
+  project_id = data.gcore_project.project.id
+}
+
+resource "gcore_volume" "boot_volume_windows" {
+  name       = "windows boot volume"
+  type_name  = "ssd_hiiops"
+  size       = 50
+  image_id   = data.gcore_image.windows.id
+  project_id = data.gcore_project.project.id
+  region_id  = data.gcore_region.region.id
+}
+
+resource "gcore_instancev2" "instance" {
+  flavor_id     = "g1w-standard-4-8"
+  name          = "my-windows-instance"
+  password      = "my-s3cR3tP@ssw0rd"
+  user_data     = "PHBvd2Vyc2hlbGw+CiMgQmUgc3VyZSB0byBzZXQgdGhlIHVzZXJuYW1lIGFuZCBwYXNzd29yZCBvbiB0aGVzZSB0d28gbGluZXMuIE9mIGNvdXJzZSB0aGlzIGlzIG5vdCBhIGdvb2QKIyBzZWN1cml0eSBwcmFjdGljZSB0byBpbmNsdWRlIGEgcGFzc3dvcmQgYXQgY29tbWFuZCBsaW5lLgokVXNlciA9ICJTZWNvbmRVc2VyIgokUGFzc3dvcmQgPSBDb252ZXJ0VG8tU2VjdXJlU3RyaW5nICJzM2NSM3RQQHNzdzByZCIgLUFzUGxhaW5UZXh0IC1Gb3JjZQpOZXctTG9jYWxVc2VyICRVc2VyIC1QYXNzd29yZCAkUGFzc3dvcmQKQWRkLUxvY2FsR3JvdXBNZW1iZXIgLUdyb3VwICJSZW1vdGUgRGVza3RvcCBVc2VycyIgLU1lbWJlciAkVXNlcgpBZGQtTG9jYWxHcm91cE1lbWJlciAtR3JvdXAgIkFkbWluaXN0cmF0b3JzIiAtTWVtYmVyICRVc2VyCjwvcG93ZXJzaGVsbD4="
```

Comment:
let's make variable and then convert it to base64 here, because it's not obvious where this line came from
smth like: 
user_data = base64encode(var.user_data)